jwst/guider_cds/tests/test_guider_cds.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+import pytest
+
+from jwst import datamodels
+from jwst.guider_cds.guider_cds import get_dataset_info, guider_cds
+
+
+@pytest.fixture(scope='module')
+def make_guider_image():
+    """Generate science image"""
+
+    image = datamodels.GuiderRawModel()
+
+    image.meta.instrument.name = 'FGS'
+    image.meta.exposure.frame_time = 234.3423235
+    image.meta.exposure.ngroups = 4
+    image.meta.exposure.group_time = 465.643643
+
+    image.data = np.random.rand(4,10,10,10)
+
+    return image
+
+
+def test_get_dataset_info(make_guider_image):
+    """Make sure information assigned to datamodel is retrieved correctly."""
+
+    model = make_guider_image
+    model.meta.exposure.type = 'FGS_FINEGUIDE'
+
+    imshape, n_int, grp_time, exp_type = get_dataset_info(model)
+
+    assert (imshape == (10,10) and
+            n_int == model.data.shape[0] and
+            grp_time == model.meta.exposure.group_time and
+            exp_type == model.meta.exposure.type)
```

Comment:
Again, this would be better broken up into individual `assert` statements, as it will be easier to debug if something fails.

Reply:
I see, that makes sense. Changes made